index.js

```diff
@@ -1,29 +1,46 @@
 'use strict';
 var dgram = require('dgram');
-var onetime = require('onetime');
 var roots = require('root-hints')('A');
 var isReachable = require('is-reachable');
 var randomItem = require('random-item');
 var objectAssign = require('object-assign');
 var hostnames = require('./hostnames');
+var Promise = require('pinkie-promise');
 
 var timeout = 2000;
 
-module.exports = function (options, cb) {
-	if (typeof options === 'function') {
-		cb = options;
-		options = {};
-	}
 
-	options = objectAssign({ hostnames: hostnames, timeout: timeout }, options);
+module.exports = function (options) {
+	options = getOptions(options);
```

Comment:
I don't really see the benefit of having this in an external function. Just keep it inline.